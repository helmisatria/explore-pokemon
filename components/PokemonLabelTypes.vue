<template>
  <figure>
    <figcaption class="sr-only">Types</figcaption>
    <ul
      :class="variation === 'small' ? 'space-x-1' : 'space-x-2'"
      class="flex items-center"
    >
      <li
        v-for="(pokemonType, index) in types"
        :key="index"
        :class="[
          variation === 'small' ? 'text-xs px-2' : 'px-3',
          `bg-pokemon-${pokemonType.type.name}`,
        ]"
        class="border border-gray-400 text-white rounded py-[2px] capitalize"
      >
        {{ pokemonType.type.name }}
      </li>
    </ul>
  </figure>
</template>

<script>
export default {
  name: 'PokemonLabelTypes',
  props: {
    variation: {
      type: String,
      default: 'normal',
    },
    types: {
      type: Array,
      required: true,
    },
  },
}
</script>

<style></style>
