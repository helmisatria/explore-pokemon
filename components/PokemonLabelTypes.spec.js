import { createLocalVue, shallowMount } from '@vue/test-utils'
import { PiniaVuePlugin } from 'pinia'
import { createTestingPinia } from '@pinia/testing'
import PokemonLabelTypes from './PokemonLabelTypes.vue'

import pokemonDetail from '~/fixtures/pokemon-detail.json'

describe('Component: PokemonLabelTypes', () => {
  let wrapper

  const localVue = createLocalVue()
  localVue.use(PiniaVuePlugin)

  const generateWrapper = () => {
    return shallowMount(PokemonLabelTypes, {
      localVue,
      pinia: createTestingPinia(),
    })
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should mount component properly', async () => {
    await wrapper.setProps({
      types: pokemonDetail.species[0].pokemons[0].types,
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
