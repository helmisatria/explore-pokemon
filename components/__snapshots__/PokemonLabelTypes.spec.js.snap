// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonLabelTypes should mount component properly 1`] = `
<figure>
  <figcaption
    class="sr-only"
  >
    Types
  </figcaption>
   
  <ul
    class="flex items-center space-x-2"
  >
    <li
      class="text-white rounded py-[2px] capitalize px-3 bg-pokemon-fire"
    >
      
      fire
    
    </li>
  </ul>
</figure>
`;
